internal/services/network/subnet_resource.go

```diff
@@ -78,9 +78,10 @@ func resourceSubnet() *pluginsdk.Resource {
 			},
 
 			"service_endpoints": {
```

Comment:
since this is a breaking schema change, we need to make this conditionally a 3.0 thing:

```suggestion
			"service_endpoints": func() *pluginsdk.Schema {
				if !features.ThreePointOhBeta() {
					return &pluginsdk.Schema{
						Type: pluginsdk.TypeList,
						Optional: true,
 						Elem:     &pluginsdk.Schema{
 							Type: pluginsdk.TypeString,
 						},
					}
				}
				
				return &pluginsdk.Schema{
					Type: pluginsdk.TypeSet,
					Optional: true,
 					Elem:     &pluginsdk.Schema{
 						Type: pluginsdk.TypeString,
 					},
 					Set:      pluginsdk.HashString,
				}
			}(),
```

Reply:
Thanks Tom for making the suggestions. What about users who are currently using 2.x, are we going to leave things unchanged? would it be better to figure out another way than to change the schema to workaround this issue, for example, doing a sort before sending the request?